packages/eslint-plugin/src/rules/no-unsafe-call.ts

```diff
@@ -74,6 +75,52 @@ export default createRule<[], MessageIds>({
             type: isErrorType ? '`error` type' : '`any`',
           },
         });
+        return;
+      }
+
+      if (isBuiltinSymbolLike(services.program, type, 'Function')) {
+        // this also matches subtypes of `Function`, like `interface Foo extends Function {}`.
+        //
+        // For weird TS reasons that I don't understand, these are
+        //
+        // safe to construct if:
+        // - they have at least one call signature _that is not void-returning_,
+        // - OR they have at least one construct signature.
+        //
+        // safe to call (including as template) if:
+        // - they have at least one call signature
+        // - OR they have at least one construct signature.
+
+        const constructSignatures = type.getConstructSignatures();
+        if (constructSignatures.length > 0) {
+          return;
+        }
+
+        const callSignatures = type.getCallSignatures();
+        if (messageId === 'unsafeNew') {
+          const nonVoidReturnCallSignatures = callSignatures.filter(
+            signature =>
+              !tsutils.isIntrinsicVoidType(signature.getReturnType()),
+          );
+
+          if (nonVoidReturnCallSignatures.length > 0) {
+            return;
+          }
+        } else {
+          // eslint-disable-next-line no-lonely-if
```

Comment:
y?

Reply:
oh. I'll revert that.

This is just my personal vendetta; I _really_, _really_ dislike no-lonely-if. IMO if/elseif should only be used for conditions which can be safely reorded, which is incompatible with no-lonely-if